python/MDSplus/tree.py

```
@@ -777,17 +777,31 @@ def addNode(self, nodename, usage='ANY'):
                     _TreeShr._TreeSetSubtree(self.ctx, nid))
         return TreeNode(nid.value, self)
 
-    def createPulse(self, shot):
+    def createPulse(self, shot, copy_only_this=0, node_or_nid=0):
```

What about making `copy_only_this` a boolean defaulted to `False`, and then keeping the `int()` conversion down below?
Also, we might consider making `node_or_nid` default to `None`, and then converting it to 0 inside the function, to make it clear that no node is being passed, but I could go either way on that one.

---

I've implemented your suggestions, thanks. Except I left node_or_nid as 0 by default, which is iequivalent to t.getNode('\\TOP'). 

My reason for keeping node_or_nid as 0 by default is that mdsplus/tdi/treeshr/TreeCreatePulseFile.fun sets it 0 if no optional argument is supplied to the wrapper and, also, 0 is not "nothing", it is the top of the open tree. So 0 is kind of a "true" default value while "None" is misleading, in my opinion.